nautobot/extras/models/computed_fields.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+
+from django import forms
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.serializers.json import DjangoJSONEncoder
+from django.db import models
+
+# from nautobot.extras.models import ComputedField, CustomField
+# from nautobot.extras.models import ComputedField
+from django.urls import reverse
+from jinja2 import TemplateError
+
+from nautobot.core.models import BaseModel
+from nautobot.extras.models import ChangeLoggedModel
+from nautobot.extras.utils import extras_features, FeatureQuery, TemplateRenderer
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ComputedFieldManager(models.Manager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, model):
+        """
+        Return all CustomFields assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.get_queryset().filter(content_type=content_type)
+
+
+@extras_features("graphql")
+class ComputedField(BaseModel, ChangeLoggedModel):
+    """
+    A custom link to an external representation of a Nautobot object. The link text and URL fields accept Jinja2 template
+    code to be rendered with an object as context.
+    """
+
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        limit_choices_to=FeatureQuery("computed_fields"),
+    )
```

Comment:
@dgarros would probably suggest that we should have something similar to the Relationship model's `source_filter`/`destination_filter` as a way to optionally restrict the scope of this ComputedField to a subset of the given ContentType.

Reply:
I think we need both :) 
if I understand correctly this filter is to ensure that we can only create a computed field for model/content_type that was flagged with extras_features("computed_fields") 
But it would be great to extend the concept of `filter` to custom_fields and computed_fields as well. I'll open another issue for that